cumulusci/tasks/github_commit.py

```diff
@@ -0,0 +1,188 @@
+import base64
+import hashlib
+import io
+import logging
+import os
+
+from cumulusci.core.exceptions import GithubException
+
+
+class CommitDir(object):
+    """Commit all changes in local_dir to branch/repo_dir"""
+
+    def __init__(self, repo, logger=None, author={}):
+        """
+        repo: github3.py Repository object, authenticated
+        logger (optional): python logger object
+        author (optional): dict of name, email, time - default uses auth'd info
+        """
+        self.repo = repo
+        if not logger:
+            logger = logging.getLogger(os.path.basename(__file__))
+        self.logger = logger
+        self.author = author
+
+    def __call__(self, local_dir, branch, repo_dir, dry_run=False):
+        """
+        local_dir: path to local directory to commit
+        branch: target branch name
+        repo_dir: target path within repo - use '' for repo root
+        """
+
+        # get ref to branch HEAD
+        head = self.repo.ref('heads/{}'.format(branch))
+
+        # get commit pointed to by HEAD
+        commit = self.repo.git_commit(head.object.sha)
+
+        # get tree of commit
+        tree = self.repo.tree(commit.tree.sha) # shallow tree
+        tree = tree.recurse() # recurse/flatten
+        tree = tree.to_json() # convert to native types
+
+        # create new tree (delete, update)
+        new_tree_list = []
+        changed = False
```

Comment:
Sure. I hesitated to use flag at all for this, would be nice to just compare before-and-after but they will always be different due to remove of subtree items. But even so, I'm now thinking it is better to refactor and do a comparison at the end, it will be cleaner and hopefully clearer to read.